lib/arc/file.ex

```diff
@@ -70,12 +74,26 @@ defmodule Arc.File do
     case remote_file do
       {:ok, body} -> File.write(local_path, body)
       {:error, error} -> {:error, error}
+      other -> validate_arc_file(other)
+    end
+  end
+
+  defp validate_arc_file(struct) do
+    cond do
+      struct.__struct__ == Arc.File -> struct
+      true -> {:error, :invalid_data}
     end
   end
 
   defp get_remote_path(remote_path) do
     case HTTPoison.get(remote_path) do
       {:ok, %{status_code: 200, body: body}} -> {:ok, body}
+      {:ok, %{status_code: 302, headers: headers}} ->
+        cond do
+          {"Location", url} = headers |> List.keyfind("Location", 0) -> new(url)
```

Comment:
HTTPoison has a `follow_redirect: true` option we can pass.

I'd much rather use that than build our own recursive function.  For example, we're not following `301` redirects.